pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/JavaInstanceMain.java

```diff
@@ -186,6 +188,10 @@ public void run() {
                 }
             }
         });
+
+        // registering jvm metrics to prometheus
+        DefaultExports.initialize();
```

Comment:
This we need to be careful if we're running in thread mode, since it will register multiple times and throw exception. I'd avoid this for thread-based (or process based for that matter)

Reply:
JavaInstanceMain is only invoked by kubernetes and process runtime